### DISCUSSION TITLE
Guidance/advice on setting point scores for GitHub Classroom assignments with multiple tests

### DISCUSSION THREAD

**Discussion by sierrajulietromeo** [community/community#122791] [GitHub Education]
### Hi everyone,

Hi, 

I have successfully run an intro to programming module using GitHub classroom with 5 separate assignments.
It's in JS/NodeJS. I have set up and used Autograding using Jest.

Let's take the first assignment as an example. When setting up the autograding, I set it to a total of 6 points. This is because there are 6 tests.

[Here](https://github.com/sierrajulietromeo/PRG1-RPS-JS) is a public copy of the first assignment.

However, when commits by students are made to the classroom, if they pass ALL 6 tests, they get 6/6.  However if they pass 5 of the tests, GitHub classroom reports as 0/6.   I.e. they only get the points score if they pass all 6 tests. 

What tweak do I need to make GitHub classroom report 4/6 passed (and 4 points awarded), and so on? I assume it is an issue with my xxx.test.js file, or not?

Many thanks, solving this would make the next run of this module even sweeter. 👍

**Comment by fuhrmanator** [comment]
I have the same problem -- people have requested this feature in https://github.com/education/autograding/issues/146 -- but that repo is archived on May 2, 2024, with no info about what the next step is. Fingers crossed that this will exist next year.

Since I have like 48 tests spread out into multiple files, I posted an issue (with a workaround) in the node example repo that has a bash script that CoPilot wrote for me (several iterations) that will try to find the test files and generate a YAML that runs in a GitHub Action. https://github.com/classroom-resources/autograding-example-node/issues/4 
However, it doesn't really fix the issue you're talking about, because one file can still contain multiple tests. 

If you make each test be a single file, it can work. With 6 tests, you could also just define 6 test files (commands) for the assignment,, such as `npx jest test1.test.js`, `npx jest tetst2.test.js`, etc. each worth one point. However, splitting them up is probably tedious and defeats the purpose of common test setup modules, etc. You probably don't want to run `npm install` on each one (just the first one), but I didn't test it.

**Comment by sierrajulietromeo** [comment]
Thanks for your reply. Glad it's not just me. You're right, it would be overly tedious to set up with separate files, and my later assignments have more than six tests. To me, this issue seems like something that should be fixable, so fingers crossed for something soon.
